content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,223 @@
+use this environment variable automatically to connect to the remote REANA
+cluster later on. The typical value is https://reana.cern.ch for CERN deployments.
+Here is how we can set it:
+
+```python
+import os
+
+if not os.getenv('REANA_SERVER_URL'):
+    os.environ['REANA_SERVER_URL'] = 'https://reana.cern.ch'
+```
+
+In order for `reana-client` to successfully connect to remote REANA cluster
+we also need to specify another mandatory variable `REANA_ACCESS_TOKEN`. One way
+to obtain it is by opening https://reana.cern.ch/ on your browser, signing in
+with your CERN credentials and navigating to your profile. Remember to keep it
+safe and private, since this is your main key to access REANA cluster.
+Let's set it up:
+
+```python
+from getpass import getpass
+
+my_reana_token = \
+    os.getenv("REANA_ACCESS_TOKEN") or getpass('Enter your REANA token: ')
+```
+
+### Workflow specification
+
+Once all the configurations to connect to REANA cluster is set up, we can focus
+on our workflow creation. As a first step let's choose a name for the workflow
+and the kind of workflow specification we will use:
+
+```python
+my_workflow_name = 'root6-roofit'
+my_workflow_type = 'serial'
+```
+
+Now we can declare what kind of inputs our workflow will have to use.
+Please refer to [reana.yaml](https://github.com/reanahub/reana-demo-root6-roofit/blob/f29e98b482fe8cb801735ac2fa48bc01e6cc05b7/reana.yaml#L2-L9)
+for correspondence. Inputs consist of files and parameters and can be specified like this:
+
+```python
+my_inputs = {
+    'files': [
+        'code/gendata.C',
+        'code/fitdata.C'
+    ],  # A list of files your analysis will be using
+    'parameters': {
+        'events': '20000',
+        'data': 'results/data.root',
+        'plot': 'results/plot.png',
+    }  # Parameters for your workflow
+}
+```
+
+Finally we need to provide the actual workflow specification where all the
+computational steps needed to run our analysis are described. For correspondence
+please refer to [reana.yaml](https://github.com/reanahub/reana-demo-root6-roofit/blob/f29e98b482fe8cb801735ac2fa48bc01e6cc05b7/reana.yaml#L13-L22).
+The analysis will consist of two stages. In the first stage, signal and
+background are generated. In the second stage, a fit will be made for the
+signal and background. Here is how we can define the workflow steps:
+
+```python
+my_workflow = {
+    'steps': [
+        {
+            'name': 'gendata',
+            'environment': 'reanahub/reana-env-root6:6.18.04',
+            'commands': [
+                'mkdir -p results',
+                'root -b -q \'code/gendata.C(${events},"${data}")\' | tee gendata.log',
+            ],
+        },
+        {
+            'name': 'fitdata',
+            'environment': 'reanahub/reana-env-root6:6.18.04',
+            'commands': [
+                'root -b -q \'code/fitdata.C("${data}","${plot}")\' | tee fitdata.log'
+            ],
+        },
+    ]
+}
+```
+
+### Workflow creation and usage
```

Comment:
Please split "usage" into separate sections (see above), so that we can more easily link to the snippet about getting workflow status, etc.